rand_core/src/block.rs

```diff
@@ -352,27 +352,24 @@ where
 {
     #[inline]
     fn next_u32(&mut self) -> u32 {
-        let mut index = self.index * 2 - self.half_used as usize;
-        if index >= self.results.as_ref().len() * 2 {
+        let mut index = self.index - self.half_used as usize;
```

Comment:
There's an implicit contract here: we must not have `self.half_used` when `self.index == 0`. This should never be the case, but the compiler won't be able to prove that nothing else manipulates the struct, thus will have a possible-panic point here — but only in debug builds, so that should be fine.